components/script/dom/performancenavigationtiming.rs

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#![allow(unused)]
+
+use dom::bindings::codegen::Bindings::PerformanceBinding::DOMHighResTimeStamp;
+use dom::bindings::codegen::Bindings::PerformanceNavigationTimingBinding::NavigationType;
+use dom::bindings::codegen::Bindings::PerformanceNavigationTimingBinding::PerformanceNavigationTimingMethods;
+use dom::bindings::num::Finite;
+use dom::bindings::root::Dom;
+use dom::bindings::str::DOMString;
+use dom::document::Document;
+use dom::performanceresourcetiming::PerformanceResourceTiming;
+use dom_struct::dom_struct;
+
+
+#[dom_struct]
+// https://w3c.github.io/navigation-timing/#dom-performancenavigationtiming
+pub struct PerformanceNavigationTiming {
+    // https://w3c.github.io/navigation-timing/#PerformanceResourceTiming
+    performanceresourcetiming: PerformanceResourceTiming,
+    navigation_start: u64,
+    navigation_start_precise: u64,
+    document: Dom<Document>,
+    nav_type: NavigationType,
+    redirect_count: u32,
+}
+
+impl PerformanceNavigationTiming {
+    fn new_inherited(nav_start: u64,
```

Comment:
This isn't used anywhere yet. It needs to be hooked up to performanceobserver.rs, presumably.